src/common.h

```diff
@@ -100,14 +100,37 @@ typedef Bytes48 KZGProof;
 
 /** Stores the setup and parameters needed for computing KZG proofs. */
 typedef struct {
-    /** The length of `roots_of_unity`, a power of 2. */
-    uint64_t max_width;
-    /** Powers of the primitive root of unity determined by `SCALE2_ROOT_OF_UNITY` in bit-reversal
-     * permutation order, length `max_width`. */
-    fr_t *roots_of_unity;
-    /** The expanded roots of unity. */
+    /**
+     * The size of our multiplicative subgroup (the roots of unity). This is the size of the
+     * extended domain (after the RS encoding has been applied), so the size of the subgroup is
+     * FIELD_ELEMENTS_PER_EXT_BLOB.
+     */
+    uint64_t domain_size;
+    /**
+     * Roots of unity for the subgroup of size `domain_size`.
+     *
+     * The array contains `domain_size + 1` elements, it starts and ends with Fr::one().
+     */
     fr_t *expanded_roots_of_unity;
-    /** The bit-reversal permuted roots of unity. */
+    /**
+     * Roots of unity in bit-reversed order.
+     *
+     * This array is derived by applying a bit-reversal permutation to `expanded_roots_of_unity`
+     * excluding the last element. Essentially:
+     *   `brp_roots_of_unity = bit_reversal_permutation(expanded_roots_of_unity[:-1])`
+     *
+     * The array contains `domain_size` elements.
+     */
+    fr_t *brp_roots_of_unity;
+    /**
+     * Roots of unity for the subgroup of size `domain_size` in reversed order.
+     *
+     * It is the reversed version of `expanded_roots_of_unity`. Essentially:
+     *    `reverse_roots_of_unity = reverse(expanded_roots_of_unity)`
+     *
+     * This array is primarily used in FFTs.
```

Comment:
I don't feel that this remark adds all that much.

Reply:
I added the python pseudocode in an attempt to highlight the difference between this and `brp_roots_of_unity` (both in terms of "reverse" vs "reverse-bit order" and in terms of excluding the last element).

With regards to "This array is primarily used in FFTs.", I was confused on why we need the reverse roots in the first place, so this would be useful to me, but I understand that it might be superfluous.